recipes/libmysqlclient/all/conanfile.py

```diff
@@ -100,160 +75,156 @@ def loose_lt_semver(v1, v2):
-            deps_shared.extend(["KERBEROS", "SASL", "LDAP", "PROTOBUF", "CURL"])
+        deps_shared = ["SSL", "KERBEROS", "SASL", "LDAP", "PROTOBUF", "CURL"]
         for dep in deps_shared:
-            replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                  f"MYSQL_CHECK_{dep}_DLLS()",
-                                  "")
-
-        sources_cmake = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        sources_cmake_orig = os.path.join(self._source_subfolder, "CMakeListsOriginal.txt")
-        rename(self, sources_cmake, sources_cmake_orig)
-        rename(self, "CMakeLists.txt", sources_cmake)
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                            f"MYSQL_CHECK_{dep}_DLLS()",
+                            "")
+
         if self.settings.os == "Macos":
-            replace_in_file(self, os.path.join(self._source_subfolder, "libmysql", "CMakeLists.txt"),
-                "COMMAND %s" % ("$<TARGET_FILE:libmysql_api_test>" if Version(self.version) < "8.0.25" else "libmysql_api_test"),
-                "COMMAND DYLD_LIBRARY_PATH=%s %s" %(os.path.join(self.build_folder, "library_output_directory"), os.path.join(self.build_folder, "runtime_output_directory", "libmysql_api_test")))
-        replace_in_file(self, os.path.join(self._source_subfolder, "cmake", "install_macros.cmake"),
-            "  INSTALL_DEBUG_SYMBOLS(",
-            "  # INSTALL_DEBUG_SYMBOLS(")
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["DISABLE_SHARED"] = not self.options.shared
-        cmake.definitions["STACK_DIRECTION"] = "-1"  # stack grows downwards, on very few platforms stack grows upwards
-        cmake.definitions["WITHOUT_SERVER"] = True
-        cmake.definitions["WITH_UNIT_TESTS"] = False
-        cmake.definitions["ENABLED_PROFILING"] = False
-        cmake.definitions["MYSQL_MAINTAINER_MODE"] = False
-        cmake.definitions["WIX_DIR"] = False
-        if self._with_lz4:
-            cmake.definitions["WITH_LZ4"] = "system"
-
-        if self._with_zstd:
-            cmake.definitions["WITH_ZSTD"] = "system"
-            cmake.definitions["ZSTD_INCLUDE_DIR"] = self.deps_cpp_info["zstd"].include_paths[0]
+            replace_in_file(self, os.path.join(self.source_folder, "libmysql", "CMakeLists.txt"),
+                            f"COMMAND {'libmysql_api_test'}",
+                            f"COMMAND DYLD_LIBRARY_PATH={os.path.join(self.build_folder, 'library_output_directory')} {os.path.join(self.build_folder, 'runtime_output_directory', 'libmysql_api_test')}")
+        replace_in_file(self, os.path.join(self.source_folder, "cmake", "install_macros.cmake"),
+                        "  INSTALL_DEBUG_SYMBOLS(",
+                        "  # INSTALL_DEBUG_SYMBOLS(")
+
+    def generate(self):
+        cmake = CMakeToolchain(self)
```

Comment:
VirtualBuildEnv should be added since there are build requirements.

Reply:
Done, I'm still learning the ropes so thanks for checking I'm not missing things like these, I really appreciate it :)